mkosi/__init__.py

```diff
@@ -3303,6 +3303,10 @@ def copy_git_files(src: Path, dest: Path, *, source_file_transfer: SourceFileTra
         files |= {sm / x.decode("utf-8") for x in c.stdout.rstrip(b"\0").split(b"\0")}
         files -= submodules
 
+        # Add the .git submodule file well.
+        if source_file_transfer == SourceFileTransfer.copy_git_more:
+            files.add(os.path.join(sm, ".git"))
```

Comment:
Have you tested that this works for you? I just had a minute to look further up in the function and saw the dance we do to include the files in the `.git/` of the parent; so just to make sure, have you tested that the contents are included here?

Reply:
Well, the dance above is necessary because it includes all the files in the `.git/` directory of the checkout, while I have a submodule, and that is just a single file pointing to a directory within the `.git/` directory already copied before.

I was trying to do a `pip install <path-to-submodule>` and that was breaking because it couldn't get the version from git.
With this change, it works.